### DISCUSSION TITLE
Verbose logging or some way to see what's happening

### DISCUSSION THREAD

**Discussion by just-be-dev** [ast-grep/ast-grep#904] [General]
When I run `sg scan` I really have no idea what's happening. I don't know if the rules I want to run are actually being run or what files or actually being checked. It'd be nice to add some default logging to give a little more visibility into what's happen. (I do know about the interactive mode which is incredibly helpful when your rules are actually working 😅)

As an external example, when you run `deno fmt` in a project and it makes no changes it'll print (as an example) `Checked 32 files`. If a file _is_ modified, it'll print the path to each file changed before printing the number of checked files.

It would be nice if `sg scan` printed something like `Scanned 32 files for 3 rules`. It would be further helpful if there was a verbose or debug flag that printed the rules being ran and the files it was scanning for each. This could even be helpful in the single file scan mode.

**Comment by just-be-dev** [comment]
I didn't create this as an issue because there's always the possibility I just can't read and missed some flag somewhere. I did look though, promise!

**Comment by HerringtonDarkholme** [comment]
Hi! Thanks for the suggestion. I have created an issue to track the discussion in https://github.com/ast-grep/ast-grep/issues/905
